openmdao/core/problem.py

```diff
@@ -345,6 +351,18 @@ def check_partials(self, out_stream=sys.stdout, comps=None, compact_print=False,
 
         model = self.model
 
+        # Log to stdout on info channel if no logger is passed in.
+        if logger is None:
+            global _set_logger
+            if _set_logger is None:
+                logger = logging.getLogger("check_total_derivatives")
+                _set_logger = logger
+                console = logging.StreamHandler(sys.stdout)
+                console.setLevel(logging.INFO)
+                logger.addHandler(console)
+            else:
+                logger = _set_logger
+
```

Comment:
if the only diff is the logger name, that's just an arg to the function.. it is quite a bit of duplication IMO